addons/sale_purchase/tests/test_sale_purchase.py

```diff
@@ -407,3 +407,31 @@ def test_service_to_purchase_branch_tax_propagation(self):
         self.assertEqual(so.order_line.tax_ids, self.company_data['default_tax_sale'])
         so.action_confirm()
         self.assertEqual(so.order_line.purchase_line_ids.tax_ids, self.company_data['default_tax_purchase'])
+
+    def test_mto_purchase_dropship_address(self):
+        """
+        Ensure that POs for products with the MTO route have no dest_address_id as it is only used for dropshipping.
+        """
+        wh = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
+        mto_route = wh.mto_pull_id.route_id
+        mto_route.active = True
+        buy_route = wh.buy_pull_id.route_id
```

Comment:
Search directly the route:
```suggestion
        route_mto = self.env.ref('stock.route_warehouse0_mto')
        route_mto.active = True,
        route_buy = self.env.ref('purchase_stock.route_warehouse0_buy')
```